examples/src/memory/zep.ts

```diff
@@ -0,0 +1,143 @@
+import { BufferMemory } from "langchain/memory";
+import { ChatOpenAI } from "langchain/chat_models/openai";
+import { ConversationChain } from "langchain/chains";
+
+import { ZepChatMessageHistory } from "langchain/memory/";
```

Comment:
I think you have the right abstractions but are presenting the wrong interface in the example here - people will be using these in chains, which themselves take a memory object directly, not a `ChatMessageHistory`:

https://js.langchain.com/docs/modules/memory/examples/motorhead_memory#usage

Reply:
@jacoblee93 thanks. There are two ways you can use Zep Memory from Langchain:
// Option 1: Use ZepMemory directly:
import { ConversationChain } from "langchain/chains";
import { ZepMemory } from "langchain/memory/zep_memory";
const memory = new ZepMemory(sessionID, zepURL);
const chain = new ConversationChain({ llm: model, memory });

// Option 2: Use ZepChatMessageHistory with BufferMemory:
import { ConversationChain } from "langchain/chains";
import { BufferMemory } from "langchain/memory";
import { ZepChatMessageHistory } from "langchain
const memory = new BufferMemory({chatHistory: new ZepChatMessageHistory(sessionID, zepURL)});
const chain = new ConversationChain({ llm: model, memory });

I have added 2 examples, this will show up in docs under Memory as Zep Memory and Zep Chat Memory. Please advise - happy to merge them if there is some meaningful way to do so or re-organize them as appropriate.

Reply:
Ahhh ok. Was a bit confused about that. What's the difference between 1 and 2? Why not just have option 1?

The right place for`ZepChatMessageHistory` if you want it exported is with the other message stores in `src/stores/zep.ts` (same place as the Redis example).

Reply:
I think my preference unless there's a really compelling time you'd want to do option 2 over option 1 would be keeping things simple: one memory class that internally takes care of storing messages and presenting those messages to the LLM (your memory class), then one retriever class responsible for loading from a chat session.

Reply:
@jacoblee93 Yes, for now thats fine. I have removed the example reference and the related doc for Option 2.

Reply:
I have also remove the related code to make this cleaner and refactored the int tests. Thanks, hope this is everything. If you see any issues please let me know.

Reply:
Thank you both for all your hard work on this 🙏